crates/vault-registry/src/types.rs

```diff
@@ -176,17 +186,28 @@ pub struct SystemVault<Balance> {
     pub to_be_redeemed_tokens: Balance,
 }
 
-impl<AccountId: Default + Ord, BlockNumber: Default, Balance: HasCompact + Default>
-    Vault<AccountId, BlockNumber, Balance>
+impl<AccountId: Default + Ord, BlockNumber: Default, Balance: HasCompact + Default, CurrencyId>
+    Vault<AccountId, BlockNumber, Balance, CurrencyId>
 {
-    pub(crate) fn new(id: AccountId, public_key: BtcPublicKey) -> Vault<AccountId, BlockNumber, Balance> {
+    // note: public only for testing purposes
+    pub fn new(
+        id: AccountId,
+        public_key: BtcPublicKey,
+        currency_id: CurrencyId,
+    ) -> Vault<AccountId, BlockNumber, Balance, CurrencyId> {
         let wallet = Wallet::new(public_key);
         Vault {
             id,
             wallet,
             banned_until: None,
             status: VaultStatus::Active(true),
-            ..Default::default()
+            currency_id,
+            issued_tokens: Default::default(),
+            liquidated_collateral: Default::default(),
+            replace_collateral: Default::default(),
+            to_be_issued_tokens: Default::default(),
+            to_be_redeemed_tokens: Default::default(),
+            to_be_replaced_tokens: Default::default(),
```

Comment:
```suggestion
            ..Default::default()
```

Reply:
This won't work since Vault no longer implements Default